tcms/testplans/views.py

```diff
@@ -253,28 +254,32 @@ def calculate_stats_for_testplans(plans):
     return plans
 
 
-def get(request,  # pylint: disable=missing-permission-required
-        plan_id, slug=None, template_name='plan/get.html'):
-    """Display the plan details."""
+class TestPlanGetView(DetailView):  # pylint: disable=missing-permission-required
 
-    try:
-        test_plan = TestPlan.objects.select_related().get(plan_id=plan_id)
-    except ObjectDoesNotExist:
-        raise Http404
+    template_name = 'plan/get.html'
+    http_method_names = ['get']
+    model = TestPlan
 
-    if slug is None:
-        return HttpResponsePermanentRedirect(reverse('test_plan_url',
-                                                     args=[plan_id, slugify(test_plan.name)]))
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        confirmed_cases = TestCaseStatus.get_confirmed()
+        context['review_case_count'] = self.object.case.exclude(
+            case_status__name=confirmed_cases).count()
+        context['run_case_count'] = self.object.case.filter(
+            case_status__name=confirmed_cases).count()
```

Comment:
`confirmed_case` is an object which will be translated into ID when the ORM query is translated to SQL. In the way you have written it Django is smart enough to actually join on case status (the __name portion) and search for name equal to "CONFIRMED".

Here's the query (without count):
```
>>> print(self.object.case.filter(case_status__name=confirmed_case).query)
SELECT "testcases_testcase"."case_id", "testcases_testcase"."creation_date", "testcases_testcase"."is_automated", "testcases_testcase"."script", "testcases_testcase"."arguments", "testcases_testcase"."extra_link", "testcases_testcase"."summary", "testcases_testcase"."requirement", "testcases_testcase"."notes", "testcases_testcase"."text", "testcases_testcase"."case_status_id", "testcases_testcase"."category_id", "testcases_testcase"."priority_id", "testcases_testcase"."author_id", "testcases_testcase"."default_tester_id", "testcases_testcase"."reviewer_id" FROM "testcases_testcase" INNER JOIN "testcases_testcaseplan" ON ("testcases_testcase"."case_id" = "testcases_testcaseplan"."case_id") INNER JOIN "testcases_testcasestatus" ON ("testcases_testcase"."case_status_id" = "testcases_testcasestatus"."case_status_id") WHERE ("testcases_testcaseplan"."plan_id" = 1 AND "testcases_testcasestatus"."name" = CONFIRMED)
>>> 
```

Here's how it looks like when you try to match case_status directly to the object:
```

>>> print(self.object.case.filter(case_status=confirmed_case).query)
SELECT "testcases_testcase"."case_id", "testcases_testcase"."creation_date", "testcases_testcase"."is_automated", "testcases_testcase"."script", "testcases_testcase"."arguments", "testcases_testcase"."extra_link", "testcases_testcase"."summary", "testcases_testcase"."requirement", "testcases_testcase"."notes", "testcases_testcase"."text", "testcases_testcase"."case_status_id", "testcases_testcase"."category_id", "testcases_testcase"."priority_id", "testcases_testcase"."author_id", "testcases_testcase"."default_tester_id", "testcases_testcase"."reviewer_id" FROM "testcases_testcase" INNER JOIN "testcases_testcaseplan" ON ("testcases_testcase"."case_id" = "testcases_testcaseplan"."case_id") WHERE ("testcases_testcaseplan"."plan_id" = 1 AND "testcases_testcase"."case_status_id" = 2)
```

Problem here is that your resulting queries are less efficient b/c they make a JOIN which isn't needed. Rewrite to compare case_status directly.

Reply:
Note: we could also argue if using an intermediate variable (which produces +1 DB query) will have negligible impact on performance but we'll not b/c this needs to be measured.

For reference the resulting query if I don't use an intermediate variable is still the same:
```
>>> print(self.case.filter(case_status=TestCaseStatus.get_confirmed()).query)
SELECT "testcases_testcase"."case_id", "testcases_testcase"."creation_date", "testcases_testcase"."is_automated", "testcases_testcase"."script", "testcases_testcase"."arguments", "testcases_testcase"."extra_link", "testcases_testcase"."summary", "testcases_testcase"."requirement", "testcases_testcase"."notes", "testcases_testcase"."text", "testcases_testcase"."case_status_id", "testcases_testcase"."category_id", "testcases_testcase"."priority_id", "testcases_testcase"."author_id", "testcases_testcase"."default_tester_id", "testcases_testcase"."reviewer_id" FROM "testcases_testcase" INNER JOIN "testcases_testcaseplan" ON ("testcases_testcase"."case_id" = "testcases_testcaseplan"."case_id") WHERE ("testcases_testcaseplan"."plan_id" = 1 AND "testcases_testcase"."case_status_id" = 2)
```